disabled-tests-inspector/src/main/java/io/quarkus/ts/DisabledTestAnalyserService.java

```diff
@@ -0,0 +1,217 @@
+import jakarta.enterprise.context.ApplicationScoped;
+
+import org.kohsuke.github.GitHub;
+import org.kohsuke.github.GHRepository;
+import org.kohsuke.github.GHTree;
+import org.kohsuke.github.GHTreeEntry;
+import org.kohsuke.github.GHContent;
+import org.kohsuke.github.GHIssue;
+import org.kohsuke.github.GHIssueState;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.HashMap;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@ApplicationScoped
+public class DisabledTestAnalyserService {
+
+    private static final Pattern CLASS_DECLARATION_PATTERN = Pattern.compile(
+            "public\\s+(?:\\w+\\s+)*class\\s+(\\w+)"
+    );
+
+    private static final Pattern TEST_METHOD_PATTERN = Pattern.compile(
+            "public\\s+void\\s+(\\w+)\\s*\\("
+    );
+
+    private static final Pattern DISABLED_ANNOTATION_PATTERN = Pattern.compile(
+            "@(Disabled\\w*)\\s*(?:\\((.*)\\))?"
+    );
+
+    private static final Pattern REASON_PATTERN = Pattern.compile(
+            "reason\\s*=\\s*\"([^\"]+)\"", Pattern.CASE_INSENSITIVE
+    );
+
+    private static final Pattern EXPLICIT_REASON_PATTERN = Pattern.compile(
+            "\"(.*)\""
+    );
+
+    private static final Pattern ISSUE_PATTERN = Pattern.compile(
+    "(https://(?:github\\.com/.+?/issues/\\d+|issues\\.redhat\\.com/browse/\\w+-\\d+))"
+    );
+
+
+    public void analyzeRepository(String repoOwner, String repoName, List<String> branches, String baseOutputFileName) throws IOException {
+        GitHub github = GitHub.connect();
+        GHRepository repo = github.getRepository(repoOwner + "/" + repoName);
+
+        for (String branch : branches) {
+            GHTree tree = repo.getTreeRecursive(branch, 1);
+            List<DisabledTest> disabledTests = new ArrayList<>();
+            Map<String, DisabledTestsModuleStats> moduleStats = new HashMap<>();
+
+            for (GHTreeEntry entry : tree.getTree()) {
+                String filePath = entry.getPath();
+                if (entry.getPath().contains("/test/") && filePath.endsWith(".java")) {
+                    GHContent fileContent = repo.getFileContent(filePath, branch);
+                    disabledTests.addAll(extractDisabledTests(fileContent, moduleStats));
+                }
+            }
+
+            BranchAnalysisResult result = new BranchAnalysisResult(branch, disabledTests);
+            String testListFile = getStatsOrTestFileName(baseOutputFileName, branch, false);
+            new ObjectMapper().writerWithDefaultPrettyPrinter().writeValue(new File(testListFile), result);
+
+            String statsListFile = getStatsOrTestFileName(baseOutputFileName, branch, true);
+            new ObjectMapper().writerWithDefaultPrettyPrinter().writeValue(new File(statsListFile), moduleStats);
+        }
+    }
+
+    private List<DisabledTest> extractDisabledTests(GHContent ghContent,
```

Comment:
this whole menthod should be independent on the githup-api dependency and properly tested with unit tested. there are no unit tests in this PR but a lot of code that if you change, you will have to re-test manually every scenario.